firstvoices/backend/views/search/custom_search_views.py

```diff
@@ -0,0 +1,45 @@
+from rest_framework import mixins, viewsets
+from rest_framework.response import Response
+
+from backend.views.search.search_query_builder import get_search_query
+from backend.views.search.utils import hydrate_objects
+
+
+class CustomSearchViewSet(mixins.ListModelMixin, viewsets.GenericViewSet):
+    http_method_names = ["get"]
+    queryset = ""
+
+    def get_search_params(self):
+        """
+        Function to return search params in a structured format.
+        """
+        input_q = self.request.GET.get("q", "")
+
+        return {"q": input_q}
+
+    def get_raw_objects(self):
+        """
```

Comment:
Since this is actually executing the search, I think it deserves a more descriptive name. Maybe search(), or get_search_results() or get_search_hits()

Reply:
After removing the for loop above, there wasn't much going on in the function except for executing the query and returning the hits. I thus moved the two lines into the list function.